src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
```

Comment:
```suggestion
export function groupByKey<T>(items: T[], key: key of T): GroupsMap<T> {
```
It would be better to check key from your T type